### PR TITLE
docs: Update magnetic field documentation

### PR DESCRIPTION
**Pull request by paulgessinger**
Updates the magnetic field documentation page, and also removes the obsolete `SharedBField`.

### CODE REVIEW

File: docs/core/magnetic_field.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+{class}`Acts::AtlasStepper`, {class}`Acts::EigenStepper`) accept the magnetic
+field as an explicit argument. 
+
+## Provider interface
+
+All magnetic field implementations inherit and implement the interface {class}`Acts::MagneticFieldProvider`:
+
+
+It provides a generic interface over different implementations. To speed up
+magnetic field lookup, each implementation can have a *cache* object, which the
```
Comment (tboldagh) [inline docs/core/magnetic_field.md:18]: can explicitly say when this is useful (i.e. in checking the field in adjacent /nearby area)?

File: docs/core/magnetic_field.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+[](#magnetic-field-context) for details.
+
+The main lookup method of {class}`Acts::MagneticFieldProvider` is
+
+:::{doxygenfunction} Acts::MagneticFieldProvider::getField
+:outline:
+:::
+
+Aside from the lookup position
+as a global vector, it accepts an instance of the opaque cache object mentioned
```
Comment (tboldagh) [inline docs/core/magnetic_field.md:38]: global position vector (I mean the vector can be local variable - right?)

File: docs/core/magnetic_field.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+The magnetic field context is an opaque type which contains experiment specific
+event context information. This can be used to supply event dependent data to
+the magnetic field instance, in case it is needed to provide correct field
+values. The library itself does not make any assumptions on the content of this
+context type (it is implemented using `std::any`), but passes a reference
+through the call-chain to the field implementation. An experiment specific
+field implementation is then expected to performa cast to the concrete type,
+and use the contents. 
+
+An example of this is a conditions database handle that is derived from the
```
Comment (tboldagh) [inline docs/core/magnetic_field.md:75]: This sounds very Atlas. Maybe?:
An example of use of context could be to look up conditions database/records for the value of magnetic field in the time wen event was taken.

File: docs/core/magnetic_field.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+### Interpolated magnetic field
+
+For more complex magnetic field implementations
+{class}`Acts::InterpolatedMagneticField` can be used. The idea here is to calculate
+an interpolated value of the magnetic field from a grid of known field values.
+In 3D, this means the interpolation is done from the 8 cornerpoints of a *field
+cell*. The field cell can be retrieved for any given position. Since during
+typical access patterns, e.g. the propagation, subsequent steps are relatively
+likely to not cross the field cell boundary, the field cell can be cached.
```
Comment (tboldagh) [inline docs/core/magnetic_field.md:101]: be cached. -> be cached and thus cell lookup avoided which improves performance.

Comment (tboldagh) [reply docs/core/magnetic_field.md:101]: ah, maybe not, it is said just few lines later.

File: docs/core/magnetic_field.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+{class}`Acts::InterpolatedBFieldMap`, which is a template class that depends on
+an instance of {class}`Acts::detail::Grid`. Varying configurations are possible,
+like a 2D field map that exploits $rz$ symmetry, or a plain 3D grid.
+
+:::{doxygenclass} Acts::InterpolatedBFieldMap
+:members: false
+:::
+
+
+The class constructor accepts a single object of type configuration struct that
```
Comment (tboldagh) [inline docs/core/magnetic_field.md:124]: suggest to drop: `object of type`

Comment (paulgessinger) [reply docs/core/magnetic_field.md:124]:
```suggestion
The class constructor accepts a single configuration struct that
```

File: docs/core/magnetic_field.md (hunk 2 above)
Comment (paulgessinger) [reply docs/core/magnetic_field.md:38]: Yeah, I'm referring to the coordinate system.

Comment (paulgessinger) [reply docs/core/magnetic_field.md]: Changed.

File: docs/core/magnetic_field.md (hunk 1 above)
Comment (paulgessinger) [reply docs/core/magnetic_field.md]: Added a line on this.

**Comment by paulgessinger** [comment]
Thanks for the read, @tboldagh. I think I addressed all of your points. If you're happy, can you approve the PR?

**Review by tboldagh — APPROVED** [review]

**Pull request merged**